### PR TITLE
adding tests for components package

### PR DESCRIPTION
**Pull request by mohamedasifs123**
#### What type of PR is this?
/kind feature

#### What this PR does / why we need it:
adding tests for components package
#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
none
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs
none
```

### CODE REVIEW

File: pkg/kwokctl/components/dashboard_metrics_scraper_test.go [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+		t.Run(tt.name, func(t *testing.T) {
+			gotComponent, err := BuildDashboardMetricsScraperComponent(tt.args.conf)
+			if (err != nil) != tt.wantErr {
+				t.Errorf("BuildDashboardMetricsScraperComponent() error = %v, wantErr %v", err, tt.wantErr)
+				return
+			}
+			a := fmt.Sprintf("%v", gotComponent)
+			if !reflect.DeepEqual(a, tt.wantComponent) {
+				t.Errorf("BuildDashboardMetricsScraperComponent() = %v, want %v", gotComponent, a)
+			}
```
Comment (wzshiming) [inline pkg/kwokctl/components/dashboard_metrics_scraper_test.go:76]: Use structured want data

**Review by wzshiming — COMMENTED** [review]
Make sure that the test style is the same.

File: go.mod [hunk 2]
```diff
@@ -20,6 +20,7 @@ require (
 	github.com/prometheus/client_model v0.5.0
 	github.com/spf13/cobra v1.8.0
 	github.com/spf13/pflag v1.0.5
+	github.com/stretchr/testify v1.8.4
```
Comment (wzshiming) [inline go.mod:23]: This package is currently not expected to be imported

Comment (mohamedasifs123) [reply go.mod:23]: Okay , I will refactor the code without using this package.

**Pull request closed without merging**